src/controller/datacontroller.js

```diff
@@ -198,10 +198,28 @@ export default class DataController {
 			this.model.selection.removeAllRanges();
 			this.model.selection.clearAttributes();
 
+			// Parse data to model and extract markers from parsed document fragment.
+			const { modelItem, markersData } = this.parse( data );
```

Comment:
I was thinking a while about the return format. I don't like the code bellow needed to translate `markersData` to `document.makers`. It's not short nor simple, and you will have to write it every time you need a custom handling of parsed data. Of course, we could have a helper function, but where-ever we put such function it will be ugly. Then I thought that maybe markers should be returned as `MarkersCollection`.

At the same time, it's strange that you can not return one object representing a parsed fragment of the document, something like... `DocumentFragment`?

Markers are integrated part of the document data. They are not like selection or history, they are more like attributes or children, so maybe we should have `DocumentFragment.makers`? Then, when you add document fragment to another document, makers from it should be taken and added to the parent document, the same way like children are moved to the parent element. It could be done by the `model.writer` or `element.insert` (I hope soon we will get rid off `model.writer` anyway).